### PR TITLE
SUP-2634: game implementation

### PR DESCRIPTION
**Pull request by illia-kachkaiev**

### CODE REVIEW

File: scene.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
+package main
+
+import (
+	"github.com/veandco/go-sdl2/sdl"
+	"log"
+	"time"
+)
+
+type scene struct {
```
Comment (illia-kachkaiev) [inline scene.go:9]: Is it possible in Go to create a structure slice field like `renderableObjects [ ]*renderableObjectInterface` that contains different structures that implement this interface? And then iterate over this field calling common methods? 
When I tried to do so a lot of errors popped up :(

File: scene_test.go [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"github.com/veandco/go-sdl2/sdl"
+	"testing"
+)
+
+func TestNewSceneCreated(t *testing.T) {
```
Comment (illia-kachkaiev) [inline scene_test.go:8]: Is there a way in Go to check in tests if the object is instance of some struct? 
I've tried `ok := object.(someStruct)`, but seems like it doesn't work

**Review by camila-reyes — APPROVED** [review]

File: scene.go (hunk 1 above)
Comment (closmarfer) [reply scene.go:9]: Something like this?
```go
type renderableObjectInterface interface {
	fooMethod() string
}
type renderableObjects []renderableObjectInterface

type implementation struct {
}

func (i implementation) fooMethod() string {
	return "bar"
}

func foo() {
	bar := renderableObjects{}
	for _, obj := range bar {
		result := obj.fooMethod()
		fmt.Println(result)
	}
}
```

You can't use `*` when the type is an interface:
```go
renderableObjects [ ]renderableObjectInterface
```

File: scene_test.go (hunk 2 above)
Comment (closmarfer) [reply scene_test.go:8]: Something like this:
```go
func assert(argument interface{}) {
	obj, ok := argument.(renderableObjectInterface)
	if !ok {
		panic("foo")
	}
	obj.fooMethod()
}

func assert2(argument renderableObjectInterface) {
	obj, ok := argument.(renderableObjectInterface)
	if !ok {
		panic("foo")
	}
	obj.fooMethod()
}
```

Comment (illia-kachkaiev) [reply scene_test.go:8]: So it works only with interfaces, but doesn't work with a struct, I see now. Thanks!

File: scene.go (hunk 1 above)
Comment (illia-kachkaiev) [reply scene.go:9]: Yes, like this, thanks!

File: scene_test.go (hunk 2 above)
Comment (closmarfer) [reply scene_test.go:8]: Maybe it fails due to is a pointer type.
```go
type imAnObject struct {
}

func (i imAnObject) fooMethod() string {
	return "bar"
}

func isAnObject(argument interface{}) bool {
	_, ok := argument.(imAnObject)
	return ok
}

func printResult(result bool) {
	if result {
		fmt.Println("Yes is object")
	} else {
		fmt.Println("Not Object")

	}
}

func TestAssert(t *testing.T) {
	t.Run("fail with a string", func(t *testing.T) {
		printResult(isAnObject("holi"))
	})

	t.Run("fail with a pointer of object", func(t *testing.T) {
		printResult(isAnObject(&imAnObject{}))
	})

	t.Run("OK passing the object", func(t *testing.T) {
		printResult(isAnObject(imAnObject{}))
	})
}

```

```
=== RUN   TestAssert
=== RUN   TestAssert/fail_with_a_string
Not Object
=== RUN   TestAssert/fail_with_a_pointer_of_object
Not Object
=== RUN   TestAssert/OK_passing_the_object
Yes is object
--- PASS: TestAssert (0.00s)
    --- PASS: TestAssert/fail_with_a_string (0.00s)
    --- PASS: TestAssert/fail_with_a_pointer_of_object (0.00s)
    --- PASS: TestAssert/OK_passing_the_object (0.00s)
PASS
```

Comment (illia-kachkaiev) [reply scene_test.go:8]: Thank you a lot, that's very helpful. This is exactly what I lacked!

**Review by closmarfer — APPROVED** [review]

**Pull request merged**